lib/npg_pipeline/product/release.pm

```diff
@@ -0,0 +1,293 @@
+package npg_pipeline::product::release;
+
+use namespace::autoclean;
+
+use Data::Dump qw[pp];
+use Moose::Role;
+
+use npg_qc::mqc::outcomes;
+
+with qw{WTSI::DNAP::Utilities::Loggable
+        npg_pipeline::base::config};
+
+our $VERSION = '0';
+
+Readonly::Scalar my $RELEASE_CONFIG_FILE => 'product_release.yml';
+
+has 'qc_schema' =>
+  (isa        => 'npg_qc::Schema',
+   is         => 'ro',
+   required   => 1,);
+
+has 'release_config' =>
+  (isa        => 'HashRef',
+   is         => 'rw',
+   required   => 1,
+   default    => sub { return {} },);
+
+sub BUILD {
+  my ($self) = @_;
+
+  my $file = $self->conf_file_path($RELEASE_CONFIG_FILE);
+
+  $self->info("Reading product release configuration from '$file'");
+  $self->release_config($self->read_config($file));
+  $self->debug('Loaded product release configuration: ',
+               pp($self->release_config));
+  return;
+}
+
+=head2 is_release_data
+
+  Arg [1]    : npg_pipeline::product
+
+  Example    : $obj->is_release_data($product)
+  Description: Return true if the product is data for release i.e.
+
+                - is not control data
+                - is not data from tag zero that could not be deplexed
+
+  Returntype : Bool
+
+=cut
+
+sub is_release_data {
+  my ($self, $product) = @_;
+
+  $product or $self->logconfess('A product argument is required');
+
+  my $rpt = $product->rpt_list();
+  my $name = $product->file_name_root();
+  if ($product->is_tag_zero_product) {
+    $self->info("Product $name, $rpt is NOT for release (is tag zero)");
+    return 0;
+  }
+
+  if ($product->lims->is_control) {
+    $self->info("Product $name, $rpt is NOT for release (is control)");
+    return 0;
+  }
+
+  $self->info("Product $name, $rpt is for release ",
+              '(is not tag zero or control)');
+
+  return 1;
+}
+
+=head2 has_qc_for_release
+
+  Arg [1]    : npg_pipeline::product
+
+  Example    : $obj->has_qc_for_release($product)
+  Description: Return true if the product has passed all QC necessary
+               to be released.
+
+  Returntype : Bool
+
+=cut
+
+sub has_qc_for_release {
+  my ($self, $product) = @_;
+
+  $product or $self->logconfess('A product argument is required');
+
+  my $rpt  = $product->rpt_list();
+  my $name = $product->file_name_root();
+  my $outcomes = npg_qc::mqc::outcomes->new(qc_schema => $self->qc_schema);
+  if ($outcomes->get_library_outcome($rpt)) {
+    $self->info("Product $name, $rpt is for release (passed manual QC)");
+    return 1;
+  }
+
+  $self->info("Product $name, $rpt is NOT for release (failed manual QC)");
```

Comment:
False value returned by get_library_outcome() does not mean 'fail' in manual qc. Just means that the product cannot be released